tests/components/sonos/test_switch.py

```diff
@@ -45,3 +46,31 @@ async def test_alarm_attributes(hass, config_entry, config, soco):
     assert alarm_state.attributes.get(ATTR_VOLUME) == 0.25
     assert alarm_state.attributes.get(ATTR_PLAY_MODE) == "SHUFFLE_NOREPEAT"
     assert not alarm_state.attributes.get(ATTR_INCLUDE_LINKED_ZONES)
+
+
+async def test_alarm_create_delete(
+    hass, config_entry, config, soco, alarm_clock, alarm_clock_extended, alarm_event
+):
+    """Test for correct creation and deletion of alarms during runtime."""
+    await setup_platform(hass, config_entry, config)
+
+    soco.alarmClock = alarm_clock_extended
+    speaker = hass.data[DATA_SONOS].discovered[soco.uid]
+    speaker.async_dispatch_alarms(event=alarm_event)
+    await hass.async_block_till_done()
+
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    assert "15" in hass.data[DATA_SONOS].alarms
+    assert "14" in hass.data[DATA_SONOS].alarms
+    assert "switch.sonos_alarm_14" in entity_registry.entities
+    assert "switch.sonos_alarm_15" in entity_registry.entities
+
+    soco.alarmClock = alarm_clock
```

Comment:
This looks weird. What's the aim here? I don't think we can replace this mock without resubscribing the subscriptions of the speaker, and expect the rest of the test to work.

If I comment out the rest of the test from here, there's no error.

Reply:
I think this works at the moment as we're not testing the whole thing. But it's better to just adjust the alarm list return value if that's what we want to change.